src/System.IO.Pipelines.Compression/Inflater.cs

```diff
@@ -0,0 +1,225 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Security;
+
+namespace System.IO.Pipelines.Compression
+{
+    /// <summary>
+    /// Provides a wrapper around the ZLib decompression API
+    /// </summary>
+    internal sealed class Inflater : IDisposable
```

Comment:
Yes, but we need to design clean low level APIs.

---
In reply to: [88018514](https://github.com/dotnet/corefxlab/pull/980#discussion_r88018514) [](ancestors = 88018514)